src/authenticator/formAuthenticator.js

```diff
@@ -83,36 +83,30 @@ FormAuthenticator.prototype.get = function(url){
     this._getField(this.conditionalLogonButtonSelector).click().then(function () {
       that.logger.debug('Opening conditional IDP auth page');
     });
-    this._waitForField(this.passFieldSelector, 'redirect to conditional auth page');
+    this._waitForField(this.passFieldSelector, 'Wating to redirect to conditional auth page');
   }
 
   this._getField(this.passFieldSelector).sendKeys(this.pass);
   this._getField(this.logonButtonSelector).click();
 
-  function _waitForAuthorizationButtonEnabled() {
-    return that._getField(that.authorizationButtonSelector).isEnabled().then(function (enabled) {
-      that.logger.debug("Is authorizationButtonSelector enabled: (" + enabled + ")");
-      return enabled;
-    }).catch(function () {
-      that.logger.debug("authorizationButtonSelector cannot be found!");
-      return true;
-    })
-  }
-
-  // handle conditional login:
+  // handle reauthorization in github
   // first a user credentials are entered and sign in button is pressed
   // then you authorize the oAuth application
   if (this.authorizationButtonSelector) {
-    this._wait(_waitForAuthorizationButtonEnabled);
-    this._getField(this.authorizationButtonSelector)
-      .click()
-      .then(function () {
-        that.logger.debug('Clicking authorizationButtonSelector.');
-      })
-      .catch(function () {
-        that.logger.debug("authorizationButtonSelector cannot be found!");
+    //this._wait(_waitForAuthorizationButtonEnabled);
+    this._wait(function() {
+      return that._isPresent(that.authorizationButtonSelector).then(function(isPresent) {
+        if (isPresent) {
+          return that._getField(that.authorizationButtonSelector).click().then(function() {
+            return true;
+          });
+        }
       });
+    },'Explicit authorization was not requested',this.authorizationButtonTimeout).catch(function () {
+      // swallow the timeout error as reauthorization is only displayed sometimes
```

Comment:
wouldn't it happen every time because the browser won't keep the session

Reply:
when Github  detects too much UI logins in a short period